app/logic/fileHandler.py

```diff
@@ -12,41 +12,55 @@ def __init__(self,files=None, courseId=None, eventId=None):
             print("Directory exists.")
 
-    def getFileFullPath(self, newfile = None):
+    def getFileFullPath(self, newfilename = ''):
         """
         This creates the directory/path for the object from the "Choose File" input in the create event and edit event.
         :returns: directory path for attachment
         """
+
+        # Added the eventID of the first recurring event to track the file path for subsequent recurring events.
         try:
             # tries to create the full path of the files location and passes if
             # the directories already exist or there is no attachment
-            filePath=(os.path.join(self.path, newfile.filename))
+            filePath=(os.path.join(self.path, "".join(newfilename)))  # This line adds the eventID of the first recurring event.
         except AttributeError:  # will pass if there is no attachment to save
             pass
         except FileExistsError:
             pass
+
         return filePath
 
-    def saveFiles(self):
-        """ Saves the attachment in the app/static/files/eventattachments/ or courseattachements/ directory """
+    def saveFiles(self, saveOriginalFile=None):
+        """ Saves the attachment in the app/static/files/eventatta chments/ or courseattachements/ directory """
         try:
             for file in self.files:
                 if self.eventId:
-                    isFileInEvent = AttachmentUpload.select().where(AttachmentUpload.event == self.eventId, AttachmentUpload.fileName == file.filename).exists()
-                    if not isFileInEvent:
-                        AttachmentUpload.create(event = self.eventId, fileName = file.filename)
-                        file.save(self.getFileFullPath(newfile = file)) # saves attachment in directory
+                    if saveOriginalFile and saveOriginalFile.id == self.eventId:       # Checks if iterant is the first (recurring) event
+                        isFileInEvent = AttachmentUpload.select().where(AttachmentUpload.event == self.eventId, AttachmentUpload.fileName == file.filename).exists()
+                        if not isFileInEvent:
+                            AttachmentUpload.create(event = self.eventId, fileName = str(saveOriginalFile.id) + "/" + file.filename)
+                            # saves attachment in the same directory using the eventID of the first recurring event.
+                            file.save(self.getFileFullPath(newfilename = str(saveOriginalFile.id) + "/" + file.filename))
+                            # We need to add the event id to the file path when we call it in file.save.
+                    else:
+                        isFileInEvent = AttachmentUpload.select().where(AttachmentUpload.event == self.eventId, AttachmentUpload.fileName == file.filename).exists()
+                        if not isFileInEvent:
+                            AttachmentUpload.create(event = self.eventId, fileName = str(saveOriginalFile.id) + "/" + file.filename)
```

Comment:
We should be able to refactor and simplify this. Right now we duplicate the `select()` and `create()` calls.